Rollback of components branch

fidsusj opened the pull request:
As we agreed on, I will split up pull request #73 
See issue #85

fidsusj commented on de.dashup.model/src/main/resources/de/dashup/model/builder/panels/1-large.html:
```diff
@@ -1,13 +1 @@
-
-</dashup-grid-layout>
\ No newline at end of file
+<dashup-weather></dashup-weather>
```
I still do not get why I should do this? This takes us the possibility to add extra logic to our widgets and adds extra DOM content to the light DOM. The other way around, our widgets would be more encapsulated and more powerful (or it would be much easier to add extra features at least). I can change it if you feel better that way, but do not complain if the users are bored from the widgets in the end.

joshuaschu replied:
We should do this because we should not create more powerful widgets than the user will be able to create. I think the user is way more bored when he sees that there are so powerful widgets and he can just create very basic things. 
In addition to that I think we should handle our widgets the same way as we handle the user widgets so there is one single style and way for all widgets.
This was a result of a team discussion so you cant change that by your own.

fidsusj replied:
This wasn't really the result of a team discussion, but rather how you interpreted the result. But never mind, if the team wants it then this is the direction we will go to. What do you think @raphaelmue @svenleonhard ?
I just do not see the benefit for our users, this just restricts the capabilities that dashup could have. Of course, users would not be able to create widgets like we do, but they can open feature requests at any time. Rendering, styling etc. would still be the same for dashup widgets and user widgets.

fidsusj commented:
> Here is my review:
> We discussed that there should NOT be a component representing a panel. See comment.
> Checked out in intellij... You have not corrected any warning in the js files. Why are we doing reviews when you do not correct the errors afterwards?? I guess this are some beers again.
> 
> Started the application. Logged in as testuser. The panels looks good but there is a bug in the todo list: when adding multiple elements and selecting some and clicking on delete the hole list is deleted.
> In addition I think the share panel is not working. If I pass Siemens as value nothing happens. If I pass SIE nothing happens. Maybe nothing happens at all???

See the upper comment regarding the widget design issue.

All of the warnings are caused by the Polymer Framework and the imported LitElement library. The syntax used in the tagged template literals will be parsed to native HTML5 by the tagged html function. IntelliJ does not take this into account. Furthermore, the functions like render are unresolved because they will be added at runtime by the LitElement base class import. (If you try it out in WebStorm after importing LitElement from npm, you should not see a single warning ;) (let me know if it does))

Thanks for reporting the bug, though it did not delete the whole list, it kept items selected in the DOM structure, which is probably what you meant. I fixed it.

The API behind the share-banner is from the [Investors Exchange](https://iextrading.com/developer/docs/). Find the stocks, that they are allowed to trade [here](https://iextrading.com/apps/stocks/). Siemens is unfortunately not supported, but a lot of other ones!  You always need to enter the stock code.

joshuaschu commented:
> > Here is my review:
> > We discussed that there should NOT be a component representing a panel. See comment.
> > Checked out in intellij... You have not corrected any warning in the js files. Why are we doing reviews when you do not correct the errors afterwards?? I guess this are some beers again.
> > Started the application. Logged in as testuser. The panels looks good but there is a bug in the todo list: when adding multiple elements and selecting some and clicking on delete the hole list is deleted.
> > In addition I think the share panel is not working. If I pass Siemens as value nothing happens. If I pass SIE nothing happens. Maybe nothing happens at all???
> 
> See the upper comment regarding the widget design issue.
> 
> All of the warnings are caused by the Polymer Framework and the imported LitElement library. The syntax used in the tagged template literals will be parsed to native HTML5 by the tagged html function. IntelliJ does not take this into account. Furthermore, the functions like render are unresolved because they will be added at runtime by the LitElement base class import. (If you try it out in WebStorm after importing LitElement from npm, you should not see a single warning ;) (let me know if it does))
> 
> Thanks for reporting the bug, though it did not delete the whole list, it kept items selected in the DOM structure, which is probably what you meant. I fixed it.
> 
> The API behind the share-banner is from the [Investors Exchange](https://iextrading.com/developer/docs/). Find the stocks, that they are allowed to trade [here](https://iextrading.com/apps/stocks/). Siemens is unfortunately not supported, but a lot of other ones! You always need to enter the stock code. If I could create the share banner as a web component I could, for instance, implement a notification, if stocks are not supported. But building our dashup widgets only making use of our generic components, customization is not possible anymore to some degree. We should put this issue on our meeting agenda for this week.

I still see issues that are 100% NOT connected to LitElement. Just look at dashup-chart.js line 19... come on this is a missing comma....

joshuaschu commented:
I have some questions that we should discuss in our next meeting I think:
What is the advantage of LitElement? We decided not to use any more libraries unless this is really necessary. Maybe you can point that out.
Second: What about sizing of the panels? The to do list has no fixed size and is expanding with new items. How should that work?

fidsusj commented:
> > > Here is my review:
> > > We discussed that there should NOT be a component representing a panel. See comment.
> > > Checked out in intellij... You have not corrected any warning in the js files. Why are we doing reviews when you do not correct the errors afterwards?? I guess this are some beers again.
> > > Started the application. Logged in as testuser. The panels looks good but there is a bug in the todo list: when adding multiple elements and selecting some and clicking on delete the hole list is deleted.
> > > In addition I think the share panel is not working. If I pass Siemens as value nothing happens. If I pass SIE nothing happens. Maybe nothing happens at all???
> > 
> > 
> > See the upper comment regarding the widget design issue.
> > All of the warnings are caused by the Polymer Framework and the imported LitElement library. The syntax used in the tagged template literals will be parsed to native HTML5 by the tagged html function. IntelliJ does not take this into account. Furthermore, the functions like render are unresolved because they will be added at runtime by the LitElement base class import. (If you try it out in WebStorm after importing LitElement from npm, you should not see a single warning ;) (let me know if it does))
> > Thanks for reporting the bug, though it did not delete the whole list, it kept items selected in the DOM structure, which is probably what you meant. I fixed it.
> > The API behind the share-banner is from the [Investors Exchange](https://iextrading.com/developer/docs/). Find the stocks, that they are allowed to trade [here](https://iextrading.com/apps/stocks/). Siemens is unfortunately not supported, but a lot of other ones! You always need to enter the stock code. If I could create the share banner as a web component I could, for instance, implement a notification, if stocks are not supported. But building our dashup widgets only making use of our generic components, customization is not possible anymore to some degree. We should put this issue on our meeting agenda for this week.
> 
> I still see issues that are 100% NOT connected to LitElement. Just look at dashup-chart.js line 19... come on this is a missing comma....

I added the comma ;)

fidsusj commented:
> I have some questions that we should discuss in our next meeting I think:
> What is the advantage of LitElement? We decided not to use any more libraries unless this is really necessary. Maybe you can point that out.
> Second: What about sizing of the panels? The to do list has no fixed size and is expanding with new items. How should that work?

I will present you the benefits in the next meeting. But they can also be found in my [e-portfolio](https://github.com/fidsusj/Web-Components/blob/master/slides.pdf). Sizing the panels according to the size of the widget is a layout task. Gridstack would have done it by adding a scrollbar to the panel, we will need to ask Sven regarding this issue.

fidsusj commented:
Now added the remaining widgets and components. The remaining codacy issues cannot be removed as well as the remaining warnings for the above-mentioned reasons.

joshuaschu commented on de.dashup.application/src/main/webapp/components/dashup-radio-button.js:
```diff
@@ -0,0 +1,20 @@
+    }
+
+    getValue() {
+        return this.checked ? this.value : null;
```
Maybe avoid using inline ifs. I think this is not the best code style.

joshuaschu commented on de.dashup.application/src/main/webapp/components/dashup-switch.js:
```diff
@@ -0,0 +1,33 @@
+    }
+
+    getValue() {
+        return this.active ? this.valueRight : this.valueLeft;
```
Same here.

fidsusj commented on de.dashup.application/src/main/webapp/components/dashup-component.js:
```diff
@@ -0,0 +1,30 @@
+        });
+        new CSSStyleSheet().replace("@import url('https://cdnjs.cloudflare.com/ajax/libs/materialize/1.0.0/css/materialize.min.css')").then((sheet) => {
+            this.shadowRoot.adoptedStyleSheets = [...this.shadowRoot.adoptedStyleSheets, sheet];
+        });
```
the custom css variables specified in the user style sheet break through the boundaries of the shadow dom, so no need to add it here.
